rpcs3/util/sysinfo.cpp

```diff
@@ -68,14 +68,88 @@ namespace Darwin_ProcessInfo
 }
 #endif
 
-#if defined(ARCH_ARM64) && defined (_WIN32) && !defined(_MSC_VER)
-static inline u64 __readx18qword(int offset)
+namespace utils
 {
-	u64* peb = nullptr;
-	__asm__ volatile("mov %0, x18" : "=r"(peb));
-	return peb[offset];
-}
+#ifdef _WIN32
+	// Alternative way to read OS version using the registry.
+	static std::string get_fallback_windows_version()
+	{
+		// Some helpers for sanity
+		const auto read_reg_dword = [](HKEY hKey, std::string_view value_name) -> std::pair<bool, DWORD>
+		{
+			DWORD val;
+			DWORD len = sizeof(val);
+			if (ERROR_SUCCESS != RegQueryValueExA(hKey, value_name.data(), nullptr, nullptr, reinterpret_cast<LPBYTE>(&val), &len))
+			{
+				return { false, 0 };
+			}
+			return { true, val };
+		};
+
+		const auto read_reg_sz = [](HKEY hKey, std::string_view value_name) -> std::pair<bool, std::string>
+		{
+			constexpr usz MAX_SZ_LEN = 255;
+			char sz[MAX_SZ_LEN + 1];
+			DWORD sz_len = MAX_SZ_LEN;
+
+			// Safety; null terminate
+			sz[0] = 0;
+			sz[MAX_SZ_LEN] = 0;
+
+			// Read string
+			if (ERROR_SUCCESS != RegQueryValueExA(hKey, value_name.data(), nullptr, nullptr, reinterpret_cast<LPBYTE>(sz), &sz_len))
```

Comment:
What about RegGetValueA? It seems to return null terminated strings

Reply:
We lose nothing by just adding the null character.